pkg/restore/log_client.go

```diff
@@ -0,0 +1,1079 @@
+	tableBuffers map[int64]*cdclog.TableBuffer
+	grpcClis     grpcClis
+
+	tableFilter filter.Filter
+
+	// a map to store all drop schema ts, use it as a filter
+	dropTSMap sync.Map
+}
+
+// NewLogRestoreClient returns a new LogRestoreClient.
+func NewLogRestoreClient(
+	ctx context.Context,
+	restoreClient *Client,
+	startTs uint64,
+	endTS uint64,
+	tableFilter filter.Filter,
+	concurrency uint,
+	batchFlushPairs int,
+	batchFlushSize int64,
+	batchWriteKVPairs int,
+) (*LogClient, error) {
+	var err error
+	if endTS == 0 {
+		// means restore all log data,
+		// so we get current ts from restore cluster
+		endTS, err = restoreClient.GetTS(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	splitClient := NewSplitClient(restoreClient.GetPDClient(), restoreClient.GetTLSConfig())
+
+	cfg := concurrencyCfg{
+		Concurrency:       concurrency,
+		BatchFlushKVPairs: batchFlushPairs,
+		BatchFlushKVSize:  batchFlushSize,
+		BatchWriteKVPairs: batchWriteKVPairs,
+	}
+
+	lc := &LogClient{
+		restoreClient:  restoreClient,
+		splitClient:    splitClient,
+		startTS:        startTs,
+		endTs:          endTS,
+		concurrencyCfg: cfg,
+		meta:           new(LogMeta),
+		eventPullers:   make(map[int64]*cdclog.EventPuller),
+		tableBuffers:   make(map[int64]*cdclog.TableBuffer),
+		tableFilter:    tableFilter,
+	}
+	lc.grpcClis.clis = make(map[uint64]*grpc.ClientConn)
+	return lc, nil
+}
+
+func (l *LogClient) getGrpcConnLocked(ctx context.Context, storeID uint64) (*grpc.ClientConn, error) {
+	store, err := l.splitClient.GetStore(ctx, storeID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	opt := grpc.WithInsecure()
+	if l.restoreClient.tlsConf != nil {
+		opt = grpc.WithTransportCredentials(credentials.NewTLS(l.restoreClient.tlsConf))
+	}
+	ctx, cancel := context.WithTimeout(ctx, dialTimeout)
+
+	bfConf := backoff.DefaultConfig
+	bfConf.MaxDelay = gRPCBackOffMaxDelay
+	conn, err := grpc.DialContext(
+		ctx,
+		store.GetAddress(),
+		opt,
+		grpc.WithConnectParams(grpc.ConnectParams{Backoff: bfConf}),
+		grpc.WithKeepaliveParams(keepalive.ClientParameters{
+			Time:                gRPCKeepAliveTime,
+			Timeout:             gRPCKeepAliveTimeout,
+			PermitWithoutStream: true,
+		}),
+	)
+	cancel()
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+	// Cache the conn.
+	l.grpcClis.clis[storeID] = conn
+	return conn, nil
+}
+
+func (l *LogClient) getImportClient(ctx context.Context, peer *metapb.Peer) (sst.ImportSSTClient, error) {
+	l.grpcClis.mu.Lock()
+	defer l.grpcClis.mu.Unlock()
+	var err error
+
+	conn, ok := l.grpcClis.clis[peer.GetStoreId()]
+	if !ok {
+		conn, err = l.getGrpcConnLocked(ctx, peer.GetStoreId())
+		if err != nil {
+			log.L().Error("could not get grpc connect ", zap.Uint64("storeId", peer.GetStoreId()))
+			return nil, err
+		}
+	}
+	return sst.NewImportSSTClient(conn), nil
+}
+
+func (l *LogClient) tsInRange(ts uint64) bool {
+	if ts < l.startTS || ts > l.endTs {
+		return false
+	}
+	return true
```

Comment:
```suggestion
	return l.startTS <= ts && ts <= l.endTs
```

why the capitalization of the two TS are different 🤔